stream/src/main/java/org/apache/kafka/streaming/kstream/Window.java

```diff
@@ -22,16 +22,22 @@
 
 import java.util.Iterator;
 
-public interface Window<K, V> extends StateStore {
+public interface Window<K, V> {
 
-    void init(ProcessorContext context);
+    interface WindowInstance<K, V> extends StateStore {
```

Comment:
Why do we want to wrap WindowInstance inside Window? Could Window.build() be triggered multiple times in a life-cycle of a task?